components/logins/src/store.rs

```diff
@@ -74,63 +84,85 @@ impl PasswordStore {
         // sense though.
         // TODO: this is exposed to android-components consumers - we should
         // check if anyone actually calls it.
-        let scope = self.db.begin_interrupt_scope();
-        self.db.wipe(&scope)?;
+        let db = self.db.lock().unwrap();
+        let scope = db.begin_interrupt_scope();
+        db.wipe(&scope)?;
         Ok(())
     }
 
     pub fn wipe_local(&self) -> Result<()> {
-        self.db.wipe_local()?;
+        self.db.lock().unwrap().wipe_local()?;
         Ok(())
     }
 
-    pub fn reset(&self) -> Result<()> {
+    pub fn reset(self: Arc<Self>) -> Result<()> {
         // Reset should not exist here - all resets should be done via the
         // sync manager. It seems that actual consumers don't use this, but
         // some tests do, so it remains for now.
-        let engine = LoginsSyncEngine::new(&self);
+        let engine = LoginsSyncEngine::new(Arc::clone(&self));
         engine.do_reset(&EngineSyncAssociation::Disconnected)?;
         Ok(())
     }
 
     pub fn update(&self, login: Login) -> Result<()> {
-        self.db.update(login)
+        self.db.lock().unwrap().update(login)
     }
 
     pub fn add(&self, login: Login) -> Result<String> {
         // Just return the record's ID (which we may have generated).
-        self.db.add(login).map(|record| record.guid.into_string())
+        self.db
+            .lock()
+            .unwrap()
+            .add(login)
+            .map(|record| record.guid().into_string())
     }
 
-    pub fn import_multiple(&self, logins: &[Login]) -> Result<MigrationMetrics> {
-        self.db.import_multiple(logins)
+    pub fn import_multiple(&self, logins: Vec<Login>) -> Result<String> {
+        let metrics = self.db.lock().unwrap().import_multiple(&logins)?;
+        Ok(serde_json::to_string(&metrics)?)
     }
 
     pub fn disable_mem_security(&self) -> Result<()> {
-        self.db.disable_mem_security()
+        self.db.lock().unwrap().disable_mem_security()
+    }
+
+    pub fn new_interrupt_handle(&self) -> sql_support::SqlInterruptHandle {
+        self.db.lock().unwrap().new_interrupt_handle()
     }
 
     pub fn rekey_database(&self, new_encryption_key: &str) -> Result<()> {
-        self.db.rekey_database(new_encryption_key)
+        self.db.lock().unwrap().rekey_database(new_encryption_key)
     }
 
-    pub fn new_interrupt_handle(&self) -> sql_support::SqlInterruptHandle {
-        self.db.new_interrupt_handle()
+    pub fn check_valid_with_no_dupes(&self, login: &Login) -> Result<()> {
+        self.db.lock().unwrap().check_valid_with_no_dupes(&login)
     }
 
     /// A convenience wrapper around sync_multiple.
+    // Unforunately, iOS still uses this until they implement sync engine
```

Comment:
typo: missing "t" in "Unfortunately" - and s/implement sync engine/use the sync manager/ at the end.